src/Temporalio/Bridge/MetricAttributes.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Temporalio.Bridge
+{
+    /// <summary>
+    /// Core-owned metric attributes.
+    /// </summary>
+    internal class MetricAttributes : SafeHandle
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetricAttributes"/> class.
+        /// </summary>
+        /// <param name="meter">Core meter.</param>
+        /// <param name="attributes">Full attribute set.</param>
+        public MetricAttributes(MetricMeter meter, IEnumerable<KeyValuePair<string, object>> attributes)
+            : base(IntPtr.Zero, true)
+        {
+            using (var scope = new Scope())
+            {
+                unsafe
+                {
+                    var attrs = ConvertKeyValuePairs(scope, attributes);
+                    Ptr = Interop.Methods.metric_attributes_new(
+                        meter.Ptr, scope.ArrayPointer(attrs), (UIntPtr)attrs.Length);
+                    SetHandle((IntPtr)Ptr);
+                }
+            }
+        }
+
+        private unsafe MetricAttributes(Interop.MetricAttributes* ptr)
+            : base((IntPtr)ptr, true)
+        {
+            unsafe
+            {
+                Ptr = ptr;
+            }
+        }
+
+        /// <inheritdoc />
+        public override unsafe bool IsInvalid => false;
+
+        /// <summary>
+        /// Gets the pointer to the attributes.
+        /// </summary>
+        internal unsafe Interop.MetricAttributes* Ptr { get; private init; }
+
+        /// <summary>
+        /// Create a new instance with the given attributes appended.
+        /// </summary>
+        /// <param name="attributes">Attributes to append.</param>
+        /// <returns>New attributes with the given ones appended.</returns>
+        public MetricAttributes Append(IEnumerable<KeyValuePair<string, object>> attributes)
```

Comment:
This is internal code not user code. There is no exposed user code for an attribute class, so there is no value in a "merge" in internal code. It would never be called.

Reply:
`WithTags` could work in terms of merging is I suppose what I really mean, whereas just recording and passing the tags there wouldn't. `WithTags` implying you're going to keep around that new meter for a while.

Reply:
Given the .NET API currently, what would merging look like from a user POV? We don't expose `MetricAttributes` to users, we only expose `WithTags` that only ever needs to append.

Reply:
It wouldn't be any different - I mean `WithTags` can work _by_ calling merge.

Reply:
But there is no real difference between "`clone` + create `MetricsAttributes` + `merge`" and just "`clone` + `add_new_attrs`" . It's all just clone + vec extend with how I use this. I don't allow metrics attributes to be mutable at this time because I don't give users the concept of a standalone set of pre-created attributes because it's usually not worth it (and the .NET API at https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics works this way as well though they go one further and make you provide the full set every record call).